api/v1alpha1/common.go

```diff
@@ -16,10 +16,15 @@ limitations under the License.
 
 package v1alpha1
 
+// Phase describes the phase of the CustomNBImage
+// +kubebuilder:validation:Enum=Pending;Failed;Running;Succeeded;Unknown
+type Phase string
+
 const (
-	PhasePending   = "Pending"
-	PhaseFailed    = "Failed"
-	PhaseResolving = "Resolving"
-	PhaseBuilding  = "Building"
-	PhaseOk        = "Ready"
+	PhasePending   = Phase("Pending")
+	PhaseFailed    = Phase("Failed")
+	PhaseRunning   = Phase("Running")
+	PhaseBuilding  = Phase("Building")
+	PhaseSucceeded = Phase("Succeeded")
+	PhaseUnknown   = Phase("Unknown")
```

Comment:
From the same document (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) (search for `phase`) it looks like the Phase pattern is deprecated and unadvised. This is the advice for core Kubernetes types (I think), but the motivation probably apply to us as well...

> Some resources in the v1 API contain fields called phase, and associated message, reason, and other status fields. The pattern of using phase is deprecated. Newer API types should use conditions instead. Phase was essentially a state-machine enumeration field, that contradicted [system-design principles](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/principles.md#control-logic) and hampered evolution, since [adding new enum values breaks backward compatibility](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md). Rather than encouraging clients to infer implicit properties from phases, we prefer to explicitly expose the individual conditions that clients need to monitor. Conditions also have the benefit that it is possible to create some conditions with uniform meaning across all resource types, while still exposing others that are unique to specific resource types.

Reply:
ACK, the Phase pattern is kept because it give and easy overview of the state (via AggregatePhase()), I would also consider it deprecated and remove it later on...